src/proofTypes/groth16/processor/index.ts

```diff
@@ -1,40 +1,69 @@
 import {
+  Formatter,
   Groth16VerificationKey,
   Groth16VerificationKeyInput,
   Proof,
   ProofInput,
 } from '../types';
-import * as formatter from '../formatter';
 import { ProofProcessor } from '../../../types';
+import { ProofOptions } from '../../../session/types';
 
 class Groth16Processor implements ProofProcessor {
   /**
-   * Formats the zk-SNARK proof based on the curve.
+   * Dynamically selects the appropriate formatter module based on the provided library option.
+   *
+   * @param {ProofOptions} options - The proof options containing the library type.
+   * @throws {Error} If the library is unsupported or the module cannot be loaded.
+   * @returns {Object} The formatter module corresponding to the specified library.
+   */
+  private getFormatter(options: ProofOptions): Formatter {
```

Comment:
The entire options object isn't needed. The parameter can be `library`

Reply:
Agreed could just be library, can revert back next time if needed.  There may be groth16 producing systems that need more than just a library which was why i created the options object we can extend that as needed and not have to do too much refactoring.